Add stricter pinning for mapgenerator

zklaus opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Fixes conda-forge/mapgenerator-feedstock#1

<!--
Please add any other relevant info below:
-->

Mapgenerator has a unpinned matplotlib dependency. However, in the most recent release of matplotlib the `matplotlib.cbook.deprecation` API was removed, which is used by `mapgenerator`. To fix this, we should pin to `maplotlib <3.6`. 

This is described in conda-forge/mapgenerator-feedstock#1 and the build is updated in conda-forge/mapgenerator-feedstock#2.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

zklaus commented:
`show_diff.py` output:
```
noarch::mapgenerator-1.0.5-pyhd8ed1ab_0.tar.bz2
-    "matplotlib-base",
+    "matplotlib-base <3.6",
```

zklaus commented:
@conda-forge/core, ready for review.

hmaarrfk commented on recipe/gen_patch_json.py:
```diff
@@ -1440,6 +1440,10 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             _replace_pin("openlibm", "openlibm <0.8.0", deps, record)
 
+        # Retroactively pin a max version of matplotlib for mapgenerator 1.0.5
+        if record_name == "mapgenerator" and record["version"] == "1.0.5":
```
```suggestion
        if record_name == "mapgenerator" and record["version"] == "1.0.5" and record["build_number"] < 1:
```

zklaus commented:
Thanks for the review, @hmaarrfk. I incorporated your suggestion and rebased on to the current main.
`show_diff.py` output:
```
noarch::blinker-1.5-pyhd8ed1ab_0.tar.bz2
-    "python =2.7|>=3.5"
+    "python ==2.7.*|>=3.5"
noarch::mapgenerator-1.0.5-pyhd8ed1ab_0.tar.bz2
-    "matplotlib-base",
+    "matplotlib-base <3.6",
noarch::verta-0.21.0-pyhd8ed1ab_0.tar.bz2
-    "python =2.7|>=3.5",
+    "python ==2.7.*|>=3.5",
```

I suspect that the other two changes are due to a later update of the repodata.

zklaus commented:
@conda-forge/core, could somebody please have a look?
